src/components/enterprise-user-subsidy/data/utils.test.js

```diff
@@ -38,3 +40,391 @@ describe('transformRedeemablePoliciesData', () => {
+     * - Has no coupon codes
+     * - Has no coupon code requests
+     */
+    {
+      isAssignmentLearnerOnly: false,
+      redeemableLearnerCreditPolicies: {
+        redeemablePolicies: [
+          {
+            policyType: POLICY_TYPES.ASSIGNED_CREDIT,
+            learnerContentAssignments: [
+              { state: ASSIGNMENT_TYPES.ALLOCATED },
+            ],
+          },
+        ],
+        learnerContentAssignments: {
+          ...emptyRedeemableLearnerCreditPolicies.learnerContentAssignments,
+          assignments: [{ state: ASSIGNMENT_TYPES.ALLOCATED }],
+          hasAssignments: true,
+          allocatedAssignments: [{ state: ASSIGNMENT_TYPES.ALLOCATED }],
+          hasAllocatedAssignments: true,
+          assignmentsForDisplay: [{ state: ASSIGNMENT_TYPES.ALLOCATED }],
+          hasAssignmentsForDisplay: true,
+        },
+      },
+      hasCurrentEnterpriseOffers: false,
+      subscriptionPlan: {
+        isActive: false,
+      },
+      subscriptionLicense: undefined,
+      licenseRequests: [{ id: 1 }],
+      couponCodesCount: 0,
+      couponCodeRequests: [],
+    },
+    /**
+     * - `isAssignmentLearnerOnly`: false
+     * - Has assignable redeemable policy with allocated assignment
+     * - Has no other redeemable policies (auto-applied)
+     * - Has no enterprise offer
+     * - Has no active subscription plan and/or activated license
+     * - Has no subscription license request(s)
+     * - Has available coupon codes
+     * - Has no coupon code requests
+     */
+    {
+      isAssignmentLearnerOnly: false,
+      redeemableLearnerCreditPolicies: {
+        redeemablePolicies: [
+          {
+            policyType: POLICY_TYPES.ASSIGNED_CREDIT,
+            learnerContentAssignments: [
+              { state: ASSIGNMENT_TYPES.ALLOCATED },
+            ],
+          },
+        ],
+        learnerContentAssignments: {
+          ...emptyRedeemableLearnerCreditPolicies.learnerContentAssignments,
+          assignments: [{ state: ASSIGNMENT_TYPES.ALLOCATED }],
+          hasAssignments: true,
+          allocatedAssignments: [{ state: ASSIGNMENT_TYPES.ALLOCATED }],
+          hasAllocatedAssignments: true,
+          assignmentsForDisplay: [{ state: ASSIGNMENT_TYPES.ALLOCATED }],
+          hasAssignmentsForDisplay: true,
+        },
+      },
+      hasCurrentEnterpriseOffers: false,
+      subscriptionPlan: {
+        isActive: false,
+      },
+      subscriptionLicense: undefined,
+      licenseRequests: [],
+      couponCodesCount: 1,
+      couponCodeRequests: [],
+    },
+    /**
+     * - `isAssignmentLearnerOnly`: false
+     * - Has assignable redeemable policy with allocated assignment
+     * - Has no other redeemable policies (auto-applied)
+     * - Has no enterprise offer
+     * - Has no active subscription plan and/or activated license
+     * - Has no subscription license request(s)
+     * - Has no coupon codes
+     * - Has coupon code request(s)
+     */
+    {
```

Comment:
worth adding a test case where they don't have *anything* - no redeemable policy/allocated assignment, but also no other subsidy type or request?

Reply:
Added a test case below for this situation. Good callout.